### PR TITLE
EN-13311: fix delegation contract's owner delegation logs

### PR DESCRIPTION
**Pull request by bogdan-rosianu**
## Reasoning behind the pull request
- The `_init` function from delegation logs didn't have the first argument as the delegator address, so the `Address` field of the `delegate` event was not filled with the owner's address
- Added logs for `changeOwner` function 
  
## Proposed Changes
- Don't fetch the address from logs if the function is `_init`, but rather get it from the contract call input

## Testing procedure
- create a new delegation contract: https://docs.elrond.com/validators/delegation-manager/#creating-a-new-delegation-contract
- in the logs & events tab, you should see the owner's address as the `Address` field in the `delegate` log, as opposed to this:
https://explorer.elrond.com/transactions/24ab9b142606002bfb731d40e1c5265b18a164caee2680d421d6e17020866b21/logs

### CODE REVIEW

File: vm/systemSmartContracts/delegation.go [hunk 1]
```diff
@@ -991,6 +993,24 @@ func (d *delegation) changeOwner(args *vmcommon.ContractCallInput) vmcommon.Retu
 	d.eei.SetStorageForAddress(d.delegationMgrSCAddress, args.CallerAddr, []byte{})
 	d.eei.SetStorage([]byte(ownerKey), args.Arguments[0])
 
+	globalFund, err := d.getGlobalFundData()
```
Comment (sasurobert) [inline vm/systemSmartContracts/delegation.go:996]: make a new function for these 2 logEntry creations.

Comment (miiu96) [reply vm/systemSmartContracts/delegation.go:996]: done

File: vm/systemSmartContracts/delegation.go [hunk 2]
```diff
@@ -201,7 +203,7 @@ func (d *delegation) Execute(args *vmcommon.ContractCallInput) vmcommon.ReturnCo
 		return d.delegate(args)
 	case "unDelegate":
 		return d.unDelegate(args)
-	case "withdraw":
+	case withdraw:
```
Comment (iulianpascalau) [inline vm/systemSmartContracts/delegation.go:206]: :+1:

**Review by iulianpascalau — APPROVED** [review]

**Review by sasurobert — APPROVED** [review]

**Review by gabi-vuls — APPROVED** [review]
System test passed.

**Pull request merged**